modules/lang/elisp.el

```diff
@@ -8,7 +8,8 @@
 (require 'editor/company)
 
 (use-package elisp-mode
-  :straight nil
+  :ensure nil
+  :after company
```

Comment:
Surprisingly gave me a headache! I was getting a warning on init about functions being unable to be defined because of a missing symbol: company-mode. With async loading, the load order needs a bit more attention indeed, but the majority of your config worked just fine (at least from what I've seen -- I haven't tested all the features yet).

Reply:
Company is a pain, and should definitely be put quite early in the load order. I do have some hacks at https://github.com/TOTBWF/mumacs/blob/b801381684ba30ada6e4bca8287730f8606c6cdb/modules/editor/company.el#L23 for handling company `use-package` integration, so might be worth looking at.